src/pybind/mgr/cephadm/module.py

```diff
@@ -1417,7 +1418,9 @@ def _add_host(self, spec):
         :param host: host name
         """
         assert_valid_host(spec.hostname)
-        self._check_valid_addr(spec.hostname, spec.addr)
+        ip_addr = self._check_valid_addr(spec.hostname, spec.addr)
+        if spec.addr == spec.hostname and ip_addr:
+            spec.addr = ip_addr
```

Comment:
I think this will have bad side effects. 

* In case the cluster has working DNS, this will overwrite it
* In case users use /etc/hosts, this will make cephadm silently ignore /etc/hosts
* Imagine an IP change of a host: We'd then wrongly continue to use the old IP.

I think the idea is right, but we should probably not persist this information anywhere.

Reply:
I'm not sure what you mean by the last comment.  How can we not persist the IP?  The only reason this PR addresses the problem is because when we add a host we are not running on a container on that same host, so the DNS lookup won't return the loopback address.  In all other cases, we can't rely on DNS because it may return 127.0.1.1.

When it comes to resolving IPs for config files (ingress):

- I don't think it is practical to rely on DNS in general at runtime (vs 'host add' time) because the mgr can't look up the host it's running on without /etc/hosts getting in the way.
- Ditto for /etc/hosts
- IP changes are already problematic, since we will need to regenerate config files and we don't have a way to detect that.